pilot/pkg/networking/core/v1alpha3/networkfilter.go

```diff
@@ -39,6 +41,18 @@ import (
 // redisOpTimeout is the default operation timeout for the Redis proxy filter.
 var redisOpTimeout = 5 * time.Second
 
+func buildTelemetryNetworkFilters(class istionetworking.ListenerClass) []*listener.Filter {
+	filterstack := make([]*listener.Filter, 0)
+
+	// We add metadata exchange on inbound if any telemetry exists, as a client may expect metadata exchange.
+	// For outbound, we only need to set it if our own proxy has telemetry enabled.
```

Comment:
Should this still be conditional? naively, it seems like we to allow for metadata exchange for outbound calls even if we aren't collecting telemetry locally. otherwise, won't server-side telemetry potentially be incomplete?

Reply:
I think the comment is wrong but the code is right. Envoyfilter only adds to inbound: https://github.com/istio/istio/pull/35212/files/19b4c7cc957b0627d0fc3c45b8f83626202dddec#diff-6f01bd95b2effca99e0e138bfe1d57c761ec27e506a6f88ec69d4a64467c48d8L139

Reply:
Ah, lost context. cluster outbound, filter inbound, correct?

Reply:
Yeah. Let me clean up the comments and code a bit - I was confused myself